### PR TITLE
Jfoll/automate config api workspace prod

### PR DESCRIPTION
**Pull request by johnfolly**
# Babylon api workspace [test command]

## Requirement
 - organization api
 - solution api
 - deploy key: 
  ```yaml
  solution_id: ##
  organization_id: ##
  use_dedicated_event_hub_namespace: ##
  send_scenario_metadata_to_event_hub: ##
  ```

## How to test:
```bash
echo "Babylon api workspace command [Common usage]"
# Get all registered workspaces
babylon api workspace get-all
# get all registered workspaces and filter fields to retrieve only id and name
babylon api workspace get-all -f 'id,name'
# Get all registered workspaces and dump them in a file
babylon api workspace get-all -o ../test.json
# get all registered workspaces, filter fields to retrieve only id and name and dump them in a file
babylon api workspace get-all -f 'id,name' -o ../test.json
# create an workspace
export WRK=$(babylon api workspace create -n "Babylon workspace" | grep -oh "\w*w-\w*")
echo "Created workspace id : $WRK"
# get the created workspace details
babylon api workspace get $WRK
# dump the created workspace details in json file
babylon api workspace get $WRK -o ../test.json
# update the created workspace
babylon api workspace update ../test.json
# dump the created workspace details and filter fields to retrieve only id and name
babylon api workspace get $WRK -f 'id,name'
# dump the created workspace details in json file
babylon api workspace get $WRK -o ../test.json
# get the created workspace details, filter fields to retrieve only id and name name and dump them in a file
babylon api workspace get $WRK -f 'id,name' -o ../test.json
# get the current workspace set in babylon deploy config
babylon api workspace get-current
# get the current workspace set in babylon deploy config and filter fields to retrieve only id and name
babylon api workspace get-current -f 'id,name'
# dump the the current workspace set in babylon deploy in json file
babylon api workspace get-current -o ../test.json
# get the the current workspace set in babylon deploys, filter fields to retrieve only id and name name and dump them in a file
babylon api workspace get-current -f 'id,name' -o ../test.json
# delete the created workspace
babylon api workspace delete $WRK -f
# get all registered workspaces and filter fields to retrieve only id and name
babylon api workspace get-all -f 'id,name'
echo "Common error"
# cannot get workspace with this id, we'v delete it
babylon api workspace get $WRK
# cannot update an not existing workspace
babylon api workspace update tests/api_files/Workspace.yaml
# cannot get an not existing workspace
babylon api workspace get $WRK -f 'id,name'
# cannot get an not existing workspace
babylon api workspace get $WRK -o ../test.json
# cannot get an not existing workspace
babylon api workspace get $WRK -f 'id,name' -o ../test.json
# cannot get an not existing workspace
babylon api workspace get-current
# cannot get an not existing workspace
babylon api workspace get-current -f 'id,name'
# cannot get an not existing workspace
babylon api workspace get-current -o ../test.json
# cannot get an not existing workspace
babylon api workspace get-current -f 'id,name' -o ../test.json
# cannot delete an not existing workspace
babylon api workspace delete $WRK -f
# cannot create an workspace without specifying the name
babylon api workspace create
```

### CODE REVIEW

File: Babylon/groups/api/groups/workspace/commands/create.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+)
+def create(
+    env: Environment,
+    workspace_api: WorkspaceApi,
+    organization_id: str,
+    workspace_name: str,
+    send_scenario_metadata_to_event_hub: str,
+    use_dedicated_event_hub_namespace: str,
+    solution_id: str,
+    workspace_file: str,
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/create.py:84]: Optional[str]

File: Babylon/groups/api/groups/workspace/commands/create.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    select: bool,
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/create.py:85]: Optional[bool]

File: Babylon/groups/api/groups/workspace/commands/delete.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+        if converted_workspace_content["id"]:
+            workspace_id = converted_workspace_content["id"]
+        elif converted_workspace_content["workspace_id"]:
+            workspace_id = converted_workspace_content["workspace_id"]
+        else:
+            logger.error(f"Error: Could not found workspace id in {workspace_file}.")
+            return
+
+    print(workspace_id)
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/delete.py:90]: you should probably use `logger.info`

File: Babylon/groups/api/groups/workspace/commands/get.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    is_flag=True,
+    help="Should the path be relative to the working directory ?",
+    type=bool,
+)
+@option(
+    "--from-file",
+    "from_file",
+    is_flag=True,
+    help="In case the workspace id is retrieved from a file",
+)
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/get.py:61]: Don't use from file, prefer a `-f` option that will take a path
And set required=False on the argument and check if you have one of them

File: Babylon/groups/api/groups/workspace/commands/update.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+@command()
+@allow_dry_run
+@timing_decorator
+@pass_workspace_api
+@argument("workspace_file")
+@require_deployment_key("sendScenarioMetadataToEventHub", "send_scenario_metadata_to_event_hub")
+@require_deployment_key("useDedicatedEventHubNamespace", "use_dedicated_event_hub_namespace")
+@require_deployment_key("organization_id", "organization_id")
+@require_deployment_key("solution_id", "solution_id")
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/update.py:31]: Some yaml keys use snake_case, the other use camelCase, it should be uniformized

**Review by o-reo — COMMENTED** [review]
I'll test the commands tomorrow

File: Babylon/groups/api/groups/workspace/commands/delete.py [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+        )
+        if not converted_workspace_content:
+            logger.error("Error : can not get Workspace definition, please check your file")
+            return
+
+        if converted_workspace_content["id"]:
+            workspace_id = converted_workspace_content["id"]
+        elif converted_workspace_content["workspace_id"]:
+            workspace_id = converted_workspace_content["workspace_id"]
+        else:
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/delete.py:86]: You could have used
```python
workspace_id = converted_workspace_content["id"] or converted_workspace_content["workspace_id"]
if not workspace_id:
    logger.error(...)
    return
```

File: Babylon/groups/api/groups/workspace/commands/delete.py [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+        if not confirm(
+            f"You are trying to delete workspace {workspace_id} workspaces of organization {organization_id} \nDo you want to continue?"
+        ):
+            logger.info("Workspace deletion aborted.")
+            return
+
+        confirm_workspace_id = prompt("Confirm workspace id ")
+        if confirm_workspace_id != workspace_id:
+            logger.error("The workspace id you have type didn't mach with workspace you are trying to delete id")
+            return
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/delete.py:110]: I wonder if we could make this bit of code generic and put it in an utils/interactive function

File: Babylon/groups/api/groups/workspace/commands/get_current.py [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+        logger.info(pformat(retrieved_workspace))
+        return
+
+    converted_content = convert_keys_case(retrieved_workspace, underscore_to_camel)
+    with open(output_file, "w") as _f:
+        try:
+            json.dump(converted_content, _f, ensure_ascii=False)
+        except TypeError:
+            json.dump(converted_content.to_dict(), _f, ensure_ascii=False)
+    logger.info(f"Datset {workspace_id} detail was dumped on {output_file}")
```
Comment (o-reo) [inline Babylon/groups/api/groups/workspace/commands/get_current.py:81]: Missing an `a` in dataset

**Review by o-reo — COMMENTED** [review]
All good !

File: Babylon/groups/api/groups/workspace/commands/delete.py (hunk 7 above)
Comment (johnfolly) [reply Babylon/groups/api/groups/workspace/commands/delete.py:110]: Good idee ! Why not

**Review by o-reo — COMMENTED** [review]
All good :+1:

**Pull request merged**